tests/jsTests/indexSpec.js

```diff
@@ -0,0 +1,283 @@
+		<div class="ui modal">\
+			<div class="header"></div>\
+			<div class="content"></div>\
+			<div class="actions">\
+				<div class="ui cancel green button">Close</div>\
+			</div>\
+		</div>');
+		$('.ui.modal').hide();
+
+		// mock modal show call
+		spyOn($.fn, "modal").and.returnValue($('.ui.modal').show());
+		showResponse($('element'), { response: false, message: msg });
+		jasmine.clock().tick(4000);
+	});
+
+	afterEach(function() {
+		jasmine.clock().uninstall();
+	});
+
+	it("should be visible", function() {
+		expect($('.content')).toContainText(msg);
+		expect($('.header')).toContainText('Are you lost?');
+		expect($('.ui.modal')).toBeVisible();
+	});
+});
+
+/**
+ * Suite : Tests for password score
+ * Description : This is a test suite for checking password score
+ */
+describe("Tests for password score", function() {
+	it("should be visible", function() {
+		var passwords = {
+			0: {
+				'text': 'psswd',
+				'score': 0
+			},
+			1: {
+				'text': 'password',
+				'score': 2
+			},
+			2: {
+				'text': 'Password',
+				'score': 3
+			},
+			3: {
+				'text': 'password1',
+				'score': 3
+			},
+			4: {
+				'text': 'password#',
+				'score': 3
+			},
+			5: {
+				'text': 'Password1',
+				'score': 4
+			},
+			6: {
+				'text': 'Password#',
+				'score': 4
+			},
+			7: {
+				'text': 'Password#1',
+				'score': 5
+			},
+			8: {
+				'text': 'Password3#',
+				'score': 5
+			}
+		}
+		for (var k in passwords) {
+			if (passwords.hasOwnProperty(k)) {
+				var s = passwordScore(passwords[k]['text']);
+				expect(s['score']).toBe(passwords[k]['score']);
+			}
+		}
+	});
+});
+
+/**
+ * Suite : Tests for password field
+ * Description : Unit test to verify password with varying character limits to analyze strength progress bar percentage based on score
+ */
+describe("Tests for password field", function() {
+	beforeEach(function() {
+		setFixtures('<div class="field">\
+			<label>Password</label>\
+			<input type="text" name="name" id="password" value="test">\
+			<div class="ui basic red left pointing prompt label transition hide"></div>\
+		</div>\
+		<div class="field password-strength">\
+			<div id="password-strength-status" data-percent="0"></div>\
+		</div>');
+
+		spyOn($.fn, "progress").and.callFake(function() {
+			$('#password-strength-status').data('percent',arguments[0]['percent']);
+		});
+	});
+
+	it("should be visible", function() {
+
+		var passwords = {
+			0: {
+				'text': 'psswd',
+				'percent': 25
+			},
+			1: {
+				'text': 'password',
+				'percent': 45
+			},
+			2: {
+				'text': 'Password',
+				'percent': 75
+			},
+			3: {
+				'text': 'password1',
+				'percent': 75
+			},
+			4: {
+				'text': 'password#',
+				'percent': 75
+			},
+			5: {
+				'text': 'Password1',
+				'percent': 95
+			},
+			6: {
+				'text': 'Password#',
+				'percent': 95
+			},
+			7: {
+				'text': 'Password#1',
+				'percent': 95
+			},
+			8: {
+				'text': 'Password3#',
+				'percent': 95
+			}
```

Comment:
"ui-color in progress" is handled by semantic UI and bootstrap. Our task was only to check for percentage and score values and we have test suites for both of them. There we do not have to assert their presence. @MehaKaushik